libs/community/langchain_community/tools/databricks/_execution.py

```diff
@@ -174,13 +180,40 @@ def execute_function(
         parameters=parametrized_statement.parameters,
         **execute_statement_args,  # type: ignore
     )
-    status = response.status
-    assert status is not None, f"Statement execution failed: {response}"
-    if status.state != StatementState.SUCCEEDED:
-        error = status.error
+    if response.status and job_pending(response.status.state) and response.statement_id:
+        statement_id = response.statement_id
+        _logger.info("Retrying to get statement execution status...")
+        wait_time = 0
+        retry_cnt = 0
+        client_execution_timeout = int(
+            os.environ.get(
+                UC_TOOL_CLIENT_EXECUTION_TIMEOUT,
+                DEFAULT_UC_TOOL_CLIENT_EXECUTION_TIMEOUT,
+            )
+        )
+        while wait_time < client_execution_timeout:
+            wait = min(2**retry_cnt, client_execution_timeout - wait_time)
+            time.sleep(wait)
+            _logger.info(f"Retry times: {retry_cnt}")
+            response = ws.statement_execution.get_statement(statement_id)  # type: ignore
+            if response.status is None or not job_pending(response.status.state):
```

Comment:
Do we also want a circuit breaker here if the response returns as a complete to break out of the while loop?

if the response.status.state is StatementState.SUCCEEDED, we should break and resume, right?

Reply:
That is `not job_pending(response.status.state)` right? we just break